crates/p2p/src/client/peer_agnostic.rs

```diff
@@ -173,6 +173,10 @@ impl Client {
             .await
     }
 
+    /// ## Important
+    ///
+    /// `class_commitment` and `storage_commitment` is not propagated via P2P.
+    /// The caller must make sure to fill these fields with the correct values.
```

Comment:
It might be worth creating an additional header type here that excludes those fields perhaps? Not super important here, but we've shown a poor track record with optional/default fields e.g. pending header + block hash usage.

Reply:
Sure, I'm also in favor of such a variant. I didn't add it because I was pretty much sure some longer time ago there was a similar situation where we opted for the reverse so I wasn't sure what the convention in such case is. (I may be misremebering, I'm not entirely sure). But _anyhow_ (pun intended) - I'll just add it.

Reply:
The issue then was that we had a single gateway method which could return `PENDING | FULL` - which was unergonomic since all callers had to handle both. And it ended up with a lot of near-identical duplicate code.

In this scenario, its more of a delayed construction pattern - we are just missing the data to be filled in later.

We can do something like

```rust
pub struct PartialHeader(BlockHeader // note: private);

impl From<P2PHeader> for PartialHeader { // fill in the fields we do have, leave others as default }

impl PartialHeader {
    pub fn finalize(self, class_commitment: ClassCommitment, event_commitment: ..) -> BlockHeader {
        // Fill in the "missing" fields here, now you have a full type that's useable.
    }
}
```